projects/director/tasks/safety.py

```diff
@@ -182,32 +197,59 @@ def add_cmdline_args(
         parser.add_argument(
             '--include-label-cand-only',
             type='bool',
-            default=False,
+            default=True,
             help='When passing inputs to the classifier, use only the label targets if set to True.',
         )
+        parser.add_argument(
+            '--truncate-prediction-at',
+            type=int,
+            default=-1,
+            help='Truncate the prediction of the model after n words before feeding it to the classifier.',
+        )
+        parser.add_argument(
+            '--limit-classifier-examples-at',
+            type=int,
+            default=-1,
+            help='Limit the amount of classifier examples (used during training).',
+        )
+        parser.add_argument(
+            '--eval-classifier-use-cuda',
+            type=bool,
+            default=False,
+            help='Use the gpu for the eval classifier. This does not work during distributed training.',
+        )
         return parser
 
     def __init__(self, opt, shared=None):
+        super().__init__(opt, shared)
+        if opt['limit_classifier_examples_at'] > 0:
+            self.data = self.data[: opt['limit_classifier_examples_at']]
         self.include_label_cand_only = opt['include_label_cand_only']
+        self.truncate_prediction_at = opt['truncate_prediction_at']
         if opt.get('eval_classifier_model_file'):
-            from parlai.agents.reranker.classifier_reranker import ClassifierReranker
 
             if not shared:
                 self.classifier = create_agent_from_model_file(
-                    opt['eval_classifier_model_file']
+                    opt['eval_classifier_model_file'],
+                    opt_overrides={
+                        'datatype': 'valid',
+                        'no_cuda': not opt['eval_classifier_use_cuda'],
+                    },
                 )
+                self.classifier.opt.log()
             else:
-                self.classifier = shared['classifier']
+                print('Load the classifier from shared')
+                self.classifier = create_agent_from_shared(shared['classifier'])
             self.context = []
             DEFAULT_DELIM = '\n'
             self.delimiter = opt.get('delimiter', DEFAULT_DELIM)
         else:
             self.classifier = None
-        super().__init__(opt, shared)
 
     def share(self):
         shared = super().share()
-        shared['classifier'] = self.classifier
+        if self.classifier:
```

Comment:
are there instances where this is not instantiated?

Reply:
No. I removed it and added an assertion.